pymatgen/io/cif.py

```diff
@@ -943,25 +944,65 @@ def get_matching_coord(coord):
+                    match = get_matching_coord(coord)
+                    comp_d = {el: occu}
+                    if num_h > 0:
+                        comp_d["H"] = num_h
+                        self.warnings.append(
+                            "Structure has implicit hydrogens defined, "
+                            "parsed structure unlikely to be suitable for use "
+                            "in calculations unless hydrogens added."
+                        )
+                    comp = Composition(comp_d)
+                    if not match:
+                        coord_to_species[coord] = comp
+                        coord_to_magmoms[coord] = magmom
+                    else:
+                        coord_to_species[match] += comp
+                        # disordered magnetic not currently supported
+                        coord_to_magmoms[match] = None
                 else:
-                    coord_to_species[match] += comp
-                    # disordered magnetic not currently supported
-                    coord_to_magmoms[match] = None
+                    coord = (x, y, z)
+                    match = get_matching_coord(coord)
+                    comp_d = {el: 0.00000001}
+                    if num_h > 0:
+                        comp_d["H"] = num_h
+                        self.warnings.append(
+                            "Structure has implicit hydrogens defined, "
+                            "parsed structure unlikely to be suitable for use "
+                            "in calculations unless hydrogens added."
+                        )
+                    comp = Composition(comp_d)
+                    if not match:
+                        coord_to_species[coord] = comp
+                        coord_to_magmoms[coord] = magmom
+                    else:
+                        coord_to_species[match] += comp
+                        # disordered magnetic not currently supported
+                        coord_to_magmoms[match] = None
+            else:
+                if occu > 0:
+                    coord = (x, y, z)
+                    match = get_matching_coord(coord)
+                    comp_d = {el: occu}
+                    if num_h > 0:
+                        comp_d["H"] = num_h
+                        self.warnings.append(
+                            "Structure has implicit hydrogens defined, "
+                            "parsed structure unlikely to be suitable for use "
+                            "in calculations unless hydrogens added."
+                        )
+                    comp = Composition(comp_d)
+                    if not match:
+                        coord_to_species[coord] = comp
+                        coord_to_magmoms[coord] = magmom
+                    else:
+                        coord_to_species[match] += comp
+                        # disordered magnetic not currently supported
+                        coord_to_magmoms[match] = None
```

Comment:
It looks like there's a lot of code duplication between the `if` and `else` cases here. Without having looked into the details, this seems avoidable here. We aim for DRY code as much as possible.